recipes/sharedtags-greedyview.lua

```diff
@@ -0,0 +1,195 @@
+--- Provides functionality to share tags across all screens in awesome WM.
+-- @module sharedtags
+-- @author Albert Diserholt
+-- @copyright 2016 Albert Diserholt
+-- @license MIT
+
+-- Grab environment we need
+local awful = require("awful")
+local capi = {
+    screen = screen
+}
+
+local sharedtags = {
+    _VERSION = "sharedtags v1.0.0 for v4.0",
+    _DESCRIPTION = "Share tags for awesome window manager v4.0 with greedyview",
+    _URL = "https://github.com/dakling/dakling/awesome-www",
+    _LICENSE = [[
+        MIT LICENSE
+
+        Copyright (c) 2018 Albert Diserholt
+
+        Permission is hereby granted, free of charge, to any person obtaining a
+        copy of this software and associated documentation files (the "Software"),
+        to deal in the Software without restriction, including without limitation
+        the rights to use, copy, modify, merge, publish, distribute, sublicense,
+        and/or sell copies of the Software, and to permit persons to whom the
+        Software is furnished to do so, subject to the following conditions:
+
+        The above copyright notice and this permission notice shall be included in
+        all copies or substantial portions of the Software.
+
+        THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+        IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+        FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+        AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+        LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+        FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+        IN THE SOFTWARE.
+    ]]
+}
+
+-- Add a signal for each new screen, which just listens for the remove
+-- event, and moves over all tags when it happens.
+awful.screen.connect_for_each_screen(function(s)
+    -- When the screen is removed, all tags need to be moved over to an existing
+    -- screen. If they are not, accessing the tags will result in an error. It
+    -- doesn't make sense to fix the error, since clients on the now-hidden tags
+    -- will automatically be moved to a tag on a visible screen.
+    s:connect_signal("removed",function()
```

Comment:
These two signal connections can be replaced with just `capi.screen.connect_signal("removed", function(s)`.